api/store/create.js

```diff
@@ -0,0 +1,93 @@
+module.exports = createStore
+
+var _ = require('lodash')
+var request = require('request').defaults({json: true})
+
+var addRolePrivilege = require('../utils/add-role-privilege')
+var couchDbDesignDoc = require('../utils/couchdb-design-doc')
+var errors = require('../utils/errors')
+var getRoleOption = require('../utils/get-role-option')
+var hoodieAccessToCouchDbSecurity = require('../utils/hoodie-access-to-couchdb-security')
+var storeExists = require('./exists')
+var toDbId = require('../utils/to-db-id')
+
+function createStore (state, name, options) {
+  return storeExists(state, name)
+
+  .then(function (doesExist) {
+    if (doesExist) {
+      throw errors.CONFLICT
+    }
+
+    var doc = {
+      _id: toDbId(name),
+      access: {
+        read: {
+          role: []
+        },
+        write: {
+          role: []
+        }
+      }
+    }
+
+    if (options) {
+      _.concat(options.access).forEach(function (privilege) {
+        addRolePrivilege(doc.access, getRoleOption(options), privilege)
+      })
+    }
+
+    return state.metaDb.put(doc)
+
+    .then(function () {
+      var db = new state.PouchDB(name)
+      if (db.adapter !== 'http') {
+        return db.info()
+      }
+
+      return db.info().then(function () {
+        var createSecurityPromise = new Promise(function (resolve, reject) {
+          request({
+            method: 'put',
+            url: db.__opts.prefix + '/' + encodeURIComponent(name) + '/_security',
+            body: hoodieAccessToCouchDbSecurity(doc.access)
+          }, function (error, response, data) {
+            if (error) {
+              return reject(error)
+            }
+
+            if (response.statusCode >= 400) {
+              return reject(new Error(data.reason))
+            }
+
+            resolve()
+          })
+        })
+
+        var createDesignDocPromise = new Promise(function (resolve, reject) {
+          request({
+            method: 'put',
+            url: db.__opts.prefix + '/' + encodeURIComponent(name) + '/_design/hoodie-access',
```

Comment:
Just realised my fix assumes the opposite haha

So we'll likely need a mix of the two, or simply document that the authentication must be in the prefix? Would be nicer to work with all native options though.